### ISSUE TITLE
Bug: playground does not work with js files

### ISSUE THREAD

**Issue by reduckted** [typescript-eslint/typescript-eslint#12210]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

website

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&fileType=.js&code=DYUwLgBAHhC8EEYDcQ&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AIwEMnb9layAewC2wxC3RQA7p2hNI4MAF8QSoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3QAacDRh9BAKWFhREqbQCGnWAHcFFJZjFSAviCNA&tokens=false

### Repro Code

```TypeScript
let x = 1;
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/ban-ts-comment": "warn",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true,
    "allowJs": true,
    "checkJs": true
  }
}
```

### Expected Result

No errors.

### Actual Result

```
Internal error

o2</validateRulesConfig/<@https://typescript-eslint.io/sandbox/index.js:18:205539
validateRulesConfig@https://typescript-eslint.io/sandbox/index.js:18:205640
A@https://typescript-eslint.io/sandbox/index.js:18:204150
[finalizeConfig]@https://typescript-eslint.io/sandbox/index.js:96:2276
getConfigWithStatus@https://typescript-eslint.io/sandbox/index.js:18:78582
getConfig@https://typescript-eslint.io/sandbox/index.js:18:78654
_verifyWithFlatConfigArray@https://typescript-eslint.io/sandbox/index.js:99:14320
verify@https://typescript...
```

### Versions

See playground link

**Comment by reduckted** [comment]
🤔 That playground link doesn't seem to want to switch between the tsconfig and eslintrc file and doesn't report any errors. I suspect there's an internal error that's causing something to blow up.

But if you start fresh and write the tsconfig and eslint files, you'll see an error on the code tab.

**Comment by reduckted** [comment]
The actual error message is:

> Key "rules": Key "@typescript-eslint/ban-ts-comment": Could not find plugin "@typescript-eslint" in configuration.

That error message is being lost here:
https://github.com/typescript-eslint/typescript-eslint/blob/d81f8e84376cb0b8313c7a2c1dff2f5b0c344f82/packages/website/src/components/linter/createLinter.ts#L124
